--- laptop-mode-tools-1.71/usr/share/laptop-mode-tools/modules/cpufreq.orig	2017-03-04 02:02:25.558983334 +0100
+++ laptop-mode-tools-1.71/usr/share/laptop-mode-tools/modules/cpufreq	2017-03-04 02:03:04.729972792 +0100
@@ -45,11 +45,6 @@
 		CPU_IGNORE_NICE_LOAD="$BATT_CPU_IGNORE_NICE_LOAD"
 	fi
 	for THISCPU in /sys/devices/system/cpu/* ; do
-		if [ -f "$THISCPU/cpufreq/scaling_driver" ] &&
-		   [ "$(cat $THISCPU/cpufreq/scaling_driver)" = "intel_pstate" ]
-		then
-			continue # intel_pstate is a bit different - so it should be handled by its own module
-		fi
 		if [ -e $THISCPU/cpufreq/cpuinfo_min_freq ]; then
 			THIS_CPU_MAXFREQ="$CPU_MAXFREQ"
 			THIS_CPU_MINFREQ="$CPU_MINFREQ"
