--- laptop-mode-tools_1.57/usr/share/laptop-mode-tools/modules/laptop-mode~	2011-02-27 08:22:20.000000000 +0100
+++ laptop-mode-tools_1.57/usr/share/laptop-mode-tools/modules/laptop-mode	2011-06-26 12:29:14.496372260 +0200
@@ -193,7 +193,7 @@
 			set_sysctl /proc/sys/vm/laptop_mode   1
 			set_sysctl /proc/sys/vm/bdflush       "30 500 0 0 $AGE $AGE 60 20 0"
 			;;
-		"2.6")
+		"2.6"|3.*)
 			log "VERBOSE" "Adjusting 2.6 kernel parameters to enable laptop mode."
 			set_sysctl /proc/sys/vm/laptop_mode		  "$LM_SECONDS_BEFORE_SYNC"
 			set_sysctl /proc/sys/vm/dirty_writeback_centisecs "$AGE"
@@ -321,7 +321,7 @@
 			log "VERBOSE" "Adjusting 2.4 kernel parameters to disable laptop mode."
 			set_sysctl /proc/sys/vm/bdflush "30 500 0 0 $U_AGE $B_AGE 60 20 0"
 			;;
-		"2.6")
+		"2.6"|3.*)
 			log "VERBOSE" "Adjusting 2.6 kernel parameters to disable laptop mode."
 			set_sysctl /proc/sys/vm/dirty_writeback_centisecs   "$U_AGE"
 			set_sysctl /proc/sys/vm/dirty_expire_centisecs      "$B_AGE"
